features/steps/tabs.py

```diff
@@ -0,0 +1,79 @@
+# encoding: utf-8
+
+"""
+Step implementations for paragraph-related features
+"""
+
+from behave import given, then, when
+
+from docx import Document
+from docx.text.tabstop import TabStop
+from docx.shared import Inches
+
+from helpers import test_docx
+
+
+# given ===================================================
+@given('a tab_stops having {count} tab stops')
+def given_a_tab_stops_having_count_tab_stops(context, count):
+    count_int = {
+        '0': 0,
+        '3': 1
+    }[count]
+    document = Document(test_docx('tab-stops'))
+    paragraph_format = document.paragraphs[count_int].paragraph_format
+    context.tab_stops = paragraph_format.tab_stops
+
+
+# when ====================================================
+
+@when('I add a tab_stop')
+def when_i_add_a_tab_stop(context):
+    tab_stops = context.tab_stops
+    tab = tab_stops.add_tab()
+    tab.position = Inches(1.75)
+
+
+@when('I delete a tab_stop')
+def when_i_delete_a_tab_stop(context):
+    tab_stops = context.tab_stops
+    tab = tab_stops[1]
+    tab.delete()
```

Comment:
Hi Steve,

The revised feature/tabstops acceptance tests are now available for your 
review and hopefully your acceptance.

David



On 04/19/2016 05:41 PM, Steve Canny wrote:
>
> In features/steps/tabs.py 
> <https://github.com/python-openxml/python-docx/pull/261#discussion_r60324914>:
>
> > +
> > +
> > +# when ====================================================
> > +
> > +@when('I add a tab_stop')
> > +def when_i_add_a_tab_stop(context):
> > +    tab_stops = context.tab_stops
> > +    tab = tab_stops.add_tab()
> > +    tab.position = Inches(1.75)
> > +
> > +
> > +@when('I delete a tab_stop')
> > +def when_i_delete_a_tab_stop(context):
> > +    tab_stops = context.tab_stops
> > +    tab = tab_stops[1]
> > +    tab.delete()
>
> Okay, I did a little more research on the |del| statement and I 
> definitely love it a lot less now. I was under the mistaken impression 
> that |del| just directly called the |__del__()| method of the object, 
> but that's not how it works it turns out :)
>
> I'd say let's go with .remove() instead (analogous to the list method 
> of the same name):
>
> tab_stops.remove(tab_stops[1])
>
> This may actually be more useful, because it takes a TabStop object 
> reference rather than an index, so you could use it in a for loop for 
> example:
>
> for  tab_stopin  tab_stops:
>      if  i_dont_like_this(tab_stop):
>          tab_stops.remove(tab_stop)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/python-openxml/python-docx/pull/261/files/dbfce4c2185a8718c44be6ddd260be5c5e6856ff..98ba6b5433023e9d8d3d32330d089063cad8e1bd#r60324914>
>

-- 
David K. Woods, Ph.D.
Researcher, Lead Transana Developer
Wisconsin Center for Education Research
University of Wisconsin, Madison
http://www.transana.org